crates/revm/src/journaled_state.rs

```diff
@@ -556,7 +556,11 @@ impl JournaledState {
         db: &mut DB,
     ) -> Result<(&mut Account, bool), EVMError<DB::Error>> {
         Ok(match self.state.entry(address) {
-            Entry::Occupied(entry) => (entry.into_mut(), false),
+            Entry::Occupied(entry) => {
+                let account = entry.into_mut();
+                let is_cold = account.mark_warm();
```

Comment:
There is a bug here, if account is present inside `JournaledState` and was cold and marked warm, `journal_entry` of that change is not made

Reply:
great catch.